### ISSUE TITLE
[prefer-nullish-coalescing] false positive

### ISSUE THREAD

**Issue by jakubmazanec** [typescript-eslint/typescript-eslint#3275]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
Repo: https://github.com/jakubmazanec/typescript-eslint-prefer-nullish-coalescing

Clone and run:

```bash
npm i
npm run lint
```

Unfortunately I was unable to create more minimal example; this is just a snippet of the file containing error:
```typescript
<BorderedIconButton
  type="button"
  size="small"
  disabled={disabled || value <= (minValue ?? Number.NEGATIVE_INFINITY)} // <- error here
  data-testid="decrease"
  classes={{
    root: classes!.iconButton,
  }}
  onClick={onDecrease}
>
  <Minus />
</BorderedIconButton>
```

ESLint config:
```json5
{
  root: true,
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint"],
  parserOptions: {
    project: "tsconfig.json",
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2020,
    sourceType: "module",
  },
  rules: {
    "@typescript-eslint/prefer-nullish-coalescing": "error",
  },
}
```

TS config:
```json
{
  "compilerOptions": {
    "strict": true,
  },
  "files": ["index.tsx"],
}
```

**Expected Result**

No error.

**Actual Result**

```
   97:30  error  Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator  @typescript-eslint/prefer-nullish-coalescing
  124:30  error  Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator  @typescript-eslint/prefer-nullish-coalescing

✖ 2 problems (2 errors, 0 warnings)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.21.0` |
| `@typescript-eslint/parser`        | `4.21.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.23.0` |
| `node`                             | `14.16.0` |

**Comment by bradzacher** [comment]
just listing a variable name is pretty useless in terms of diagnosing your problem.
You need to provide a self-contained repro.

**Comment by jakubmazanec** [comment]
> 
> 
> just listing a variable name is pretty useless in terms of diagnosing your problem.
> You need to provide a self-contained repro.

Sorry, I accidentally posted the issue before I finished writing 🤦‍♂️ Link to repo is included.

**Comment by bradzacher** [comment]
This is working as intended - `disabled` is of type [`boolean | undefined`](https://github.com/jakubmazanec/typescript-eslint-prefer-nullish-coalescing/blob/cf02bea16181320766553c883c2670a9c0c7f088/index.tsx#L46).

It's impossible for the rule to know that the JSX property is being used similarly to a "conditional test".

You can get around this by using a disable comment, or by using a strict boolean expression: `disabled === true || value <= (minValue ?? Number.NEGATIVE_INFINITY)`
